codersdk/richparameters.go

```diff
@@ -116,3 +116,63 @@ func validationEnabled(param TemplateVersionParameter) bool {
 		param.Type == "bool" || // boolean type doesn't have any custom validation rules, but the value must be checked (true/false).
 		param.Type == "list(string)" // list(string) type doesn't have special validation, but we need to check if this is a correct list.
 }
+
+// ParameterResolver should be populated with legacy workload and rich parameter values from the previous build.  It then
+// supports queries against a current TemplateVersionParameter to determine whether a new value is required, or a value
+// correctly validates.
+type ParameterResolver struct {
+	Legacy []Parameter
+	Rich   []WorkspaceBuildParameter
```

Comment:
They're exposed so you can do

```
resolver := &codersdk.ParameterResolver{
    Legacy: legacyParams,
    Rich: richParams,
}
```

Otherwise we'd need a `NewParameterResolver()` function to initialize this struct from outside the package.

Reply:
Ok, so the only questionable thing is the location of `ParameterResolver`, but I think that I have already raised this concern in some other comment.